src/main/java/org/cactoos/scalar/Equality.java

```diff
@@ -62,22 +67,27 @@ public Equality(final T lft, final T rght) {
     public Integer value() throws Exception {
         final byte[] lft = this.left.asBytes();
         final byte[] rght = this.right.asBytes();
+        return new Ternary<>(
+            () -> lft.length == rght.length,
+            () -> Equality.compare(lft, rght),
+            () -> Integer.signum(lft.length - rght.length)
+        ).value();
+    }
+
+    /**
+     * Compare two byte arrays of the same size.
+     * @param lft Left array
+     * @param rght Right array
+     * @return Integer Comparison result
+     */
+    private static int compare(final byte[] lft, final byte[] rght) {
```

Comment:
@Vatavuk oh, man :) according to [this](http://www.elegantobjects.org/#principles) static methods are forbidden. Please refactor it to an inner class or somehow else. I think it should be possible to create an inner class that will handle only arrays with the same length.

Reply:
@proshin-roman I don't really understand this notion of forbidden private static methods. Yegor uses it in almost all his projects. It is a candidate for a new class but it does not mean that it should. I don't want to expose this behaviour anywhere else, I want it to be encapsulated in this class.

Reply:
@proshin-roman I can refactor it to a private inner class but I don't see the point...